mypy plugin crashes in Luigi >= 3.7.1

this project:

```
> cat pyproject.toml 
[tool.mypy]
plugins = ["luigi.mypy"]

> cat test.py
import luigi

class CustomPathParameter(luigi.PathParameter):
    """A PathParameter subclass that doesn't expose 'default' in its signature."""
    def __init__(self, **kwargs):
        super().__init__(**kwargs)

class MyTask(luigi.Task):
    path = CustomPathParameter()
```

crashes `luigi.mypy` on Python 3.11:

```
> ~/.local-py3.11/bin/python3.11 -m pip freeze | grep -E "(mypy|luigi)"
luigi==3.7.1
mypy==1.19.1
mypy-protobuf==5.0.0
mypy_extensions==1.1.0

> ~/.local-py3.11/bin/python3.11 -m mypy test.py --show-traceback
test.py:9: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 1.19.1
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "mypy/checkexpr.py", line 6103, in accept
  File "mypy/checkexpr.py", line 6138, in accept_maybe_cache
  File "mypy/nodes.py", line 2289, in accept
  File "mypy/checkexpr.py", line 508, in visit_call_expr
  File "mypy/checkexpr.py", line 645, in visit_call_expr_inner
  File "mypy/checkexpr.py", line 1496, in check_call_expr_with_callee_type
  File "mypy/checkexpr.py", line 1589, in check_call
  File "mypy/checkexpr.py", line 1832, in check_callable_call
  File "mypy/checkexpr.py", line 1270, in apply_function_plugin
  File "/home/dev/.local-py3.11/lib/python3.11/site-packages/luigi/mypy.py", line 117, in _task_parameter_field_callback
    default_idx = ctx.callee_arg_names.index("default")
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: 'default' is not in list
test.py:9: : note: use --pdb to drop into pdb
```

but passes on Python 3.13:

```
> python3.13 -m pip freeze | grep -E "(mypy|luigi)"
luigi==3.6.0
mypy==1.19.1
mypy-protobuf==5.0.0
mypy_extensions==1.1.0
types-mypy-extensions==1.0

> python3.13 -m mypy test.py --show-traceback
Success: no issues found in 1 source file
```

--- a/luigi/mypy.py
+++ b/luigi/mypy.py
@@ -114,7 +114,12 @@ class TaskPlugin(Plugin):
                         return base.args[0]
                     break
 
-        default_idx = ctx.callee_arg_names.index("default")
+        try:
+            default_idx = ctx.callee_arg_names.index("default")
+        except ValueError:
+            # If we couldn't infer from __new__, return Any
+            return AnyType(TypeOfAny.unannotated)
+
         default_args = ctx.args[default_idx]
 
         if default_args:

--- a/test/mypy_test.py
+++ b/test/mypy_test.py
@@ -93,3 +93,38 @@ MyTask(foo='1', bar="bar", unknown="unknown")
                 stdout,
             )  # check unknown argument
             self.assertIn("Found 3 errors in 1 file (checked 1 source file)", stdout)
+
+    def test_plugin_custom_parameter_subclass_without_default_arg(self):
+        """Test for issue #3376: Custom Parameter subclass without 'default' in __init__"""
+        if sys.version_info[:2] < (3, 8):
+            return
+
+        test_code = """
+import luigi
+
+
+class CustomPathParameter(luigi.PathParameter):
+    \"\"\"A PathParameter subclass that doesn't expose 'default' in its signature.\"\"\"
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+
+class MyTask(luigi.Task):
+    path = CustomPathParameter()
+"""
+
+        with tempfile.NamedTemporaryFile(suffix=".py") as test_file:
+            test_file.write(test_code.encode("utf-8"))
+            test_file.flush()
+            stdout, stderr, exitcode = api.run(
+                [
+                    "--no-incremental",
+                    "--cache-dir=/dev/null",
+                    "--show-traceback",
+                    "--config-file",
+                    "test/testconfig/pyproject.toml",
+                    test_file.name,
+                ]
+            )
+            self.assertEqual(exitcode, 0, f'mypy plugin error occurred:\nstdout: {stdout}\nstderr: {stderr}')
+            self.assertIn("Success: no issues found", stdout)

